### PR TITLE
Draft css styleguide

### PR DESCRIPTION
**Pull request by msecret**
Draft 0.0.1 of css guidelines

Comment on doc here: https://docs.google.com/a/gsa.gov/document/d/1t6Rjslq9I20reFF3xfW3FVeV2NWFaDKauDj_LQJC5mc/edit?usp=sharing

```
TODO:
```
- Another architecture type.
- An example of everything.
- More rationale.
- Ensure consistency.

### CODE REVIEW

File: css-styleguide.md [hunk 1]
```diff
@@ -0,0 +1,652 @@
+# Coding styleguide
+
+1. Preproocessor
```
Comment (msecret) [inline css-styleguide.md]: Make these actual inks in Markdown

File: css-styleguide.md [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+.rule {
+  margin: 3px;
+  text-align: center;
+}
+
+.another_rule {
+  margin: 3px;
+}
+
+// Bad
```
Comment (msecret) [inline css-styleguide.md:64]: Consistency: move // Bad examples before // Good

File: css-styleguide.md [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+  name: value;
+  name: value;
+}
+```
+
+- Strive to limit use of shorthand declarations to instances where you must
+explicitly set all the available values.
+```scss
+// Bad
+margin: inherit 3em;
```
Comment (msecret) [inline css-styleguide.md:94]: Line break after this, before // Good

File: css-styleguide.md [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+  font-family: 'Lucida Grande', 'Helvetica', sans-serif;
+```
+- Top-level numeric calculations should always be wrapped in parentheses
+```scss
+// Good
+.component {
+  width: (100% / 3);
+}
+
+// Bad
```
Comment (msecret) [inline css-styleguide.md:118]: Bad example before good.

File: css-styleguide.md [hunk 5]
```diff
@@ -0,0 +1,652 @@
...
+}
+```
+
+### Sorting
+* Follow ordering:
+	1. variables
+	2. @extend directives
+	3. @include directives
+	4. properties
+- For properties alphabetical order or type order, just keep the whole project consistant.
```
Comment (msecret) [inline css-styleguide.md]: Possibly need to explain type vs alphabetical order more in depth with examples.

File: css-styleguide.md [hunk 6]
```diff
@@ -0,0 +1,652 @@
...
+  @include media($sm) {
+    margin-top: $amount +10em;
+  }
+}
+```
+
+
+## Units
+### Measurements
+- Use **rem** units for font sizes with a em fallback.
```
Comment (msecret) [inline css-styleguide.md:193]: I've done this on Peace Corps and it's worked well so far, but I'd like more research done before it's implemented as a guide.

File: css-styleguide.md [hunk 7]
```diff
@@ -0,0 +1,652 @@
...
+
+
+## Naming
+- HTML elements should be in lowercase.
+```scss
+body, div {
+```
+- Classes should be lowercase.
+- Avoid camelcase.
+- Ensure things are understanbly named.
```
Comment (msecret) [inline css-styleguide.md]: "understandably"

File: css-styleguide.md [hunk 8]
```diff
@@ -0,0 +1,652 @@
...
+```scss
+@mixin clearfix {
+  &:after {
+    content: '';
+    display: table;
+    clear: both;
+  }
+}
+```
+- Use mixins for compoennts to change size.
```
Comment (msecret) [inline css-styleguide.md:453]: typo "components"

File: css-styleguide.md [hunk 9]
```diff
@@ -0,0 +1,652 @@
...
+```
+
+### Importing
+CSS rules get overridden later in the file. This means sass imports can be used to control inheritance and specificity.
+- Start with base elements.
+- Move to single nested classes and utils.
+- Move next to more specific classes, often with nesting.
+- Move next to overrides, possibly with !important rules.
+- Import alphabetically
+- Only modify import order for groups of files, not specific files.
```
Comment (msecret) [inline css-styleguide.md:546]: Having trouble explaining this.

File: css-styleguide.md [hunk 10]
```diff
@@ -0,0 +1,652 @@
...
+- Set variables for breakpoints at the top of your stylesheet. This
+  functionality is built into bourbon.
+```scss
+$sm: new-breakpoint(min-width 0 max-width 40em $sm_cols);
+```
+- Use variables to set the queries throughout so they are easy to adapt if necessary.
+- Place media queries nearest to the class they are affecting.
+- Rather the focusing on devices when deciding where to put breakpoints, focus
+on content.
+```scss
```
Comment (msecret) [inline css-styleguide.md:696]: Needs "bad" and "good" label.

File: css-styleguide.md [hunk 11]
```diff
@@ -0,0 +1,652 @@
...
+
+### Things to cover
+- Being careful with @extend -> architecture
+- Breakpoints
+- Sass output and folder structure there -> architecture
+- Mixins and functions
+- Full example
+
+## Preprocessor
+The  most supported preprocessor is sass/scss. Using this pre-processor means you'll get supported resources such as frameworks, libraries, tutorials and a comprehensive styleguide as support.
```
Comment (meiqimichelle) [inline css-styleguide.md]: Do you mean the most supported preprocessor at 18F or in general? Agree that this the most supported one at 18F, if that's what you mean, and that this is a good way to suggest that people consider using it ("this means you'll get XYZ").

**Comment by msecret** [comment]
Moving here for better organization: https://github.com/18F/frontend/pull/15

**Pull request closed without merging**